src/wymeditor/editor/dialogs.js

```diff
@@ -0,0 +1,501 @@
+        dialogWindowFeatures = options.dialogFeatures;
+    } else {
+        // Default
+        dialogWindowFeatures = [
+            "menubar=no",
+            "titlebar=no",
+            "toolbar=no",
+            "resizable=no",
+            "width=560",
+            "height=300",
+            "top=0",
+            "left=0"
+        ].join(",");
+    }
+
+    wDialog = window.open(
+        '',
+        "wymDialogWindow",
+        dialogWindowFeatures
+    );
+
+    if (
+        typeof wDialog !== "object" ||
+        wDialog.window !== wDialog
+    ) {
+        WYMeditor.console.warn("Could not create a dialog window");
+        return false;
+    }
+
+    // In the case where a dialog window already exists, it will be reused.
+    // If it is in the background, behind another window, without the following
+    // `focus` call, it will remain in the background and the user may not
+    // understand where it is. This `focus` call brings the dialog to the
+    // foreground, making reasonably sure the user notices it.
+    wDialog.focus();
+
+    // Construct the dialog
+    dialogHtml = options.dialogHtml || String() +
+        '<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" ' +
+                '"http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">' +
+        '<html dir="' + WYMeditor.DIRECTION + '">' +
+            '<head>' +
+                '<title>' + WYMeditor.DIALOG_TITLE + '</title>' +
+            '</head>' +
+            WYMeditor.DIALOG_BODY +
+        '</html>';
+
+    // HTML template replacements
+    htmlStrReplacements = [
+        {
+            placeholder: WYMeditor.DIRECTION,
+            replacement: options.direction
+        },
+        {
+            placeholder: WYMeditor.DIALOG_TITLE,
+            replacement: dialog ? wym._encloseString(dialog.title) : "Dialog"
+        },
+        {
+            placeholder: WYMeditor.DIALOG_BODY,
+            replacement: pBodyHtml || dialog.getBodyHtml.call(wym)
+        }
+    ];
+
+    // Perform HTML template replacements
+    for (i = 0; i < htmlStrReplacements.length; i++) {
+        dialogHtml = WYMeditor.Helper.replaceAllInStr(
+            dialogHtml,
+            htmlStrReplacements[i].placeholder,
+            htmlStrReplacements[i].replacement
+        );
+    }
+
+    dialogHtml = wym.replaceStrings(dialogHtml);
+
+    doc = wDialog.document;
+    doc.write(dialogHtml);
+    if (dialog && dialog.getBodyClass) {
+        jQuery(doc.body).addClass(dialog.getBodyClass.call(wym));
+    }
+    doc.close();
+
+    selectedContainer = wym.selectedContainer();
+
+    wDialog.onbeforeunload = function () {
+        wym.focusOnDocument();
+    };
+
+    // pre-init functions
+    if (jQuery.isFunction(options.preInitDialog)) {
+        options.preInitDialog(wym, wDialog);
+    }
+
+    // auto populate fields if selected container (e.g. A)
+    if (selectedContainer) {
+        jQuery(options.hrefSelector, doc).val(jQuery(selectedContainer)
```

Comment:
It would be nice if the `dialog` object itself could handle this with a call like
```
if (
       dialog &&
       typeof dialog.populateFieldsFromSelection !== "undefined"
 ) {
    dialog.populateFieldsFromSelection.call(wym);
}